### PR TITLE
Added PB to solr (already in place) and revised hierarchy and traverse g...

### PR DESCRIPTION
**Pull request by whikloj**
...raph templates.

I added these as new templates to not stomp on your initial work. 

However I have been able to index 125,000 in two days using the new hierarchy/traverse-graph templates. 
Using the object in the Sparql query as opposed to pulling all objects and then filtering on sameTerm seems to produce a considerable speed improvement.

cheers

### CODE REVIEW

File: islandora_transforms/UofM-hierarchy.xslt [hunk 1]
```diff
@@ -0,0 +1,118 @@
+<xsl:stylesheet version="1.0"
+  xmlns:xalan="http://xml.apache.org/xalan"
+  xmlns:sparql="http://www.w3.org/2001/sw/DataAccess/rf1/result"
+  xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+
+  <xsl:include href="/usr/local/fedora/tomcat/webapps/fedoragsearch/WEB-INF/classes/fgsconfigFinal/index/gsearch_solr/islandora_transforms/UofM-traverse-graph.xslt"/>
```
Comment (jordandukart) [inline islandora_transforms/UofM-hierarchy.xslt]: This include does not seem to match the pathing of where the other transforms live. That is:
`usr/local/fedora/tomcat/webapps/fedoragsearch/WEB-INF/classes/fgsconfigFinal/index/FgsIndex/islandora_transforms` versus `/usr/local/fedora/tomcat/webapps/fedoragsearch/WEB-INF/classes/fgsconfigFinal/index/gsearch_solr/islandora_transforms`

File: islandora_transforms/UofM_PB_to_solr.xslt [hunk 2]
```diff
@@ -0,0 +1,273 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+    xmlns:pb="http://www.pbcore.org/PBCore/PBCoreNamespace.html"
+    xmlns:foxml="info:fedora/fedora-system:def/foxml#" version="1.0"
+    exclude-result-prefixes="pb">
+    <xsl:output method="xml" indent="yes" encoding="UTF-8"/>
+
+    <xsl:include href="/usr/local/fedora/tomcat/webapps/fedoragsearch/WEB-INF/classes/fgsconfigFinal/index/gsearch_solr/islandora_transforms/library/xslt-string-functions.xslt"/>
```
Comment (jordandukart) [inline islandora_transforms/UofM_PB_to_solr.xslt]: File not included in the pull request?

File: islandora_transforms/UofM-hierarchy.xslt (hunk 1 above)
Comment (whikloj) [inline islandora_transforms/UofM-hierarchy.xslt]: As the actual server configuration is using the "gsearch_solr" path, should I be correcting all the files in this repository?

Comment (jordandukart) [inline islandora_transforms/UofM-hierarchy.xslt]: I mean realistically this should match your server setup so in the event that it needs to be pulled down in full the paths should match up what's there.

**Comment by jordandukart** [comment]
Tested, the hierarchy changes and it operates as still expected. For version control purposes it would be much easier to decipher the changes if they were made to the original files as opposed to creating and including the new XSLs and removing the old. As this a UoM specific fork that the institution has control over it will be fine to "stomp" over the original changes as these are viewed as a substantial improvement. 

In a similar vein, would you mind submitting a pull request to the upstream repo for the improvements to the hierarchical indexing? If not, with your permission, we'd like to include them to retain the performance improvements that you've produced.

**Comment by whikloj** [comment]
Made changes to hierarchy and traverse-graph. By upstream repo do you mean the daniel-dgi/UofM_gsearch_config?

**Comment by jordandukart** [comment]
The UofM_gsearch_config is a based off https://github.com/discoverygarden/basic-solr-config is more so what I mean.

**Comment by whikloj** [comment]
Okay, that repo is very different from our configuration. It does not use a hierarchy template to set ancestors. It does have a traverse-graph xslt in the library. Do you just want me to submit a pull request for that template change?

**Comment by jordandukart** [comment]
Ah my apologies, yes just for that one template.

**Pull request closed without merging**